com.civicrm.hrjobroles/CRM/Hrjobroles/BAO/HrJobRoles.php

```diff
@@ -47,21 +47,21 @@ public static function getContactRoles($cuid)  {
   }
 
   /**
-   * Get options for a given job roles field along with their database IDs.
+   * Get option values for specific option group.
    *
    * @param String $fieldName
    *
-   * @return Array
+   * @return array
    */
   public static function buildDbOptions($fieldName) {
     $queryParam = array(1 => array($fieldName, 'String'));
-    $query = "SELECT cpv.id, cpv.label from civicrm_option_value cpv
+    $query = "SELECT cpv.value, cpv.label from civicrm_option_value cpv
               LEFT JOIN civicrm_option_group cpg on cpv.option_group_id = cpg.id
               WHERE cpg.name = %1";
     $options = array();
     $result = CRM_Core_DAO::executeQuery($query, $queryParam);
     while ($result->fetch()) {
-      $options[] =  array( 'id'=>$result->id, 'label'=>strtolower($result->label) );
+      $options[] =  array( 'value'=>$result->value, 'label'=>strtolower($result->label) );
     }
     return $options;
   }
```

Comment:
Since now the DAO have pseudoconstants, I think you can use the buildOptions method (inherited from CRM_Core_DAO) instead of this buildDbOptions and won't have to depend on the internal structure of the option_groups/option_values table anymore. The only difference is that it has a different output format (but returns the same information).

This is the output format of buildDbOptions:
```
array (size=2)
  0 => 
    array (size=2)
      'value' => string 'Headquarters' (length=12)
      'label' => string 'headquarters' (length=12)
  1 => 
    array (size=2)
      'value' => string 'Home' (length=4)
      'label' => string 'home or home-office' (length=19)
```

And this is the output format of buildOptions:
```
array (size=2)
  'Headquarters' => string 'Headquarters' (length=12)
  'Home' => string 'Home or Home-Office' (length=19)
```